package/kedro_viz/models/experiment_tracking.py

```diff
@@ -78,11 +85,18 @@ def load_tracking_data(self, run_id: str):
             # tracking datasets do not have load methods defined yet but would be the
             # same as json loader.
             # pylint: disable=import-outside-toplevel
-            from kedro.extras.datasets import json, tracking
+            from kedro.extras.datasets import json, tracking, plotly, matplotlib
 
             tracking.JSONDataSet._load = json.JSONDataSet._load  # type: ignore
             tracking.MetricsDataSet._load = json.JSONDataSet._load  # type: ignore
-            self.runs[run_id] = self.dataset.load()
+            plotly.JSONDataSet._load =  json.JSONDataSet._load  # type: ignore
+            plotly.PlotlyDataSet._load = json.JSONDataSet._load 
+            matplotlib.MatplotlibWriter._load = matplotlib_writer_load
+
+            if isinstance(self.dataset, (PlotlyJSONDataSet, PlotlyDataSet, MatplotlibWriter)):
```

Comment:
Let's do a string check of `self.dataset_type` here like `if self.dataset_type in ["kedro.extras.datasets...", "..."]`. It's ugly and we should re-think this in future but it's safer than relying on the imports and consistent with how `get_dataset_type` works.

Reply:
In fact, because these are all plot types we could do as `if TRACKING_DATASET_GROUPS[self.dataset_type] is TrackingDatasetGroup.PLOT`. That's much nicer.

Reply:
ok, in this case i will move TRACKING_DATASET_GROUPS to model layer and import it in the repository layer